invokeai/frontend/web/src/features/system/components/SettingsModal/SettingsClearIntermediates.tsx

```diff
@@ -22,7 +23,14 @@ export default function SettingsClearIntermediates() {
   const [clearIntermediates, { isLoading: isLoadingClearIntermediates }] =
     useClearIntermediatesMutation();
 
+  const { data: queueStatus } = useGetQueueStatusQuery();
+  const hasPendingItems = queueStatus && (queueStatus.in_progress > 0 || queueStatus.pending > 0);
+
   const handleClickClearIntermediates = useCallback(() => {
+    if (hasPendingItems) {
+      return;
+    }
+
```

Comment:
@psychedelicious hi! thanks a lot for your feedback! I am quite confused as I cannot see such a dependency array. Do you mind telling me where exactly it is so that I can resolve this issue? thanks a lot in advance!!:)

Reply:
It's the array on L57, at the end of the `useCallback()`

Reply:
Thanks a lot!